### PR TITLE
Introduce 'latest' support for TerraformInstaller

### LINKED ISSUE #60: Enhancement: Allow Version: "latest" during Terraform Installer Task
image
I tried using "latest" in the Version field but the task fails, apparently it only accepts hard-coded values like picture provided.

### PR DESCRIPTION
**Pull request by favoretti**
This uses Hashi's API for determining latest terraform version. Although
they aren't super happy with this, they use it also in their Terraform
Enterprise offering ;)

Fixes #60

### CODE REVIEW

**Comment by favoretti** [comment]
@charleszipp Another smallish enhancement. Was browsing through issues, thought it might be a good idea.

File: TerraformInstaller/src/installer.ts [hunk 1]
```diff
@@ -6,10 +6,24 @@ import * as fs from 'fs';
...
 import { format } from 'util';
 const uuidV4 = require('uuid/v4');
+const fetch = require('node-fetch');
 const terraformToolName = "terraform";
 
 export async function download(inputVersion: string): Promise<string>{
-    var version = sanitizeVersion(inputVersion);
+    var latestVersion: string = "";
+
+    if(inputVersion == 'latest') {
```
Comment (charleszipp) [inline TerraformInstaller/src/installer.ts:15]: Lets toLower the inputVersion so we can sure that it doesnt fail if they put `Latest` or 'LATEST`

**Comment by charleszipp** [comment]
Updated the automation tests to use the `latest` option for this pr and ran fine. Resolved latest as 0.12.24. Will update the tests globally once we get this merged.

File: TerraformInstaller/src/installer.ts (hunk 1 above)
Comment (favoretti) [reply TerraformInstaller/src/installer.ts]: Lowercased and set default to `latest`.

**Comment by favoretti** [comment]
@charleszipp Can't see why pipeline failed now, but can this be merged or do you want me to do anything else with it?

**Comment by charleszipp** [comment]
> @charleszipp Can't see why pipeline failed now, but can this be merged or do you want me to do anything else with it?

No need to do anything else. The failure was because on the previous CI run, I had tested using 'latest' in the installer. The result is the backend state got upgraded to 0.12.24. The change was only made to the one release instance however. So when the subsequent release ran, it reverted to what I had originally for the version which was 0.12.20 which caused the failure. Rerunning with latest again fixed the issue.

After this merges, i will update the pipeline to use `latest` as the default so this wont happen again.

**Pull request merged**